### DISCUSSION TITLE
Event Rest APIs help

### DISCUSSION THREAD

**Discussion by nordanz** [InterDigitalInc/AdvantEDGE#362] [Q&A] [answered]
Hello there,

I have the following questions regarding the supported APIs and was hoping that you can point me faster to right direction:

1. Which API can I use to change the position of the UEs live? l want to move the UE from another UI and I want to update the position in AdvantEdge Sandbox
2. In middle of the run I want to ask the AdvantEdge to load another container as a new terminal (UE)?
3. In middle of the run I want to ask the AdvantEdge to load another container as a new Fog service?

At the end when I try to use swagger UI, I see the following error 
```Fetch error
NetworkError when attempting to fetch resource. https://my-platform-fqdn/auth/v1/api/AdvantEDGE Auth Service REST API
Fetch error
Possible cross-origin (CORS) issue? The URL origin (https://my-platform-fqdn) does not match the page (https://10.14.56.42). Check the server returns the correct 'Access-Control-Allow-*' headers.```

Thank you in advance.

**Comment by dilallkx** [comment, marked as answer]
Hi @nordanz,

Thanks for reaching out!

Here is the information you are requesting:
- General API information:
  - Sandbox Swagger UI available here when platform is deployed: `http://<your-advantedge-ip>/<your-sandbox>/api`
  - Sandbox APIs can be invoked directly from an external application using service-specific URI: `http://<your-advantedge-ip>/<your-sandbox>/<service>/v1/`
    - See ingress service names here: https://interdigitalinc.github.io/AdvantEDGE/docs/platform-mgmt/mgmt-cheat-sheet/#platform
- The GIS API can be used to dynamically change the position of UEs in the deployed scenario
  - Geospatial Subsystem: https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/gis/
  - GIS API: https://interdigitalinc.github.io/AdvantEDGE/docs/overview/overview-api/#gis-api
  - Use the `POST /geodata/{assetName}` endpoint with the updated location
- The Sandbox Controller API can be used to dynamically update the deployed scenario
  - Process lifecycle: https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/process-lifecycle/
  - Sandbox Controller API: https://interdigitalinc.github.io/AdvantEDGE/docs/overview/overview-api/#sandbox-api
  - Use the `POST /events/{type}` endpoint with the event type set to `SCENARIO-UPDATE`
  - You can use this API to dynamically create new terminal and fog services

Also, the error you are observing in Swagger UI is due to the configuration of your AdvantEDGE deployment.
You will need to set your platform FQDN or IP address in the deployment configuration file and then redeploy AdvantEDGE:
- Deployment configuration: https://interdigitalinc.github.io/AdvantEDGE/docs/platform-mgmt/mgmt-cheat-sheet/#deployment-configuration
- Set `ingress.host` to your platform IP address
- Redeploy AdvantEDGE and recreate your sandbox for the changes to take effect

I hope this helps you get started!
Regards.

**Reply by jblakley** [reply]
I had the same issue with the fqdn. This fixed it! Thanks.
